common/unsafe/src/test/java/org/apache/spark/unsafe/types/CollationFactorySuite.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.unsafe.types;
+
+import org.apache.spark.SparkIllegalArgumentException;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.apache.spark.sql.catalyst.util.CollationFactory.*;
+import static org.apache.spark.unsafe.types.UTF8String.*;
+
+import java.util.Arrays;
+
+public class CollationFactorySuite {
+    @Test
+    public void collationIdStability() {
+      CollationInfo collationInfo = getInstance().fetchCollationInfo(0);
+      assertEquals(collationInfo.collationName, "UCS_BASIC");
+      assertTrue(collationInfo.isBinaryCollation);
+
+      collationInfo = getInstance().fetchCollationInfo(1);
+      assertEquals(collationInfo.collationName, "UCS_BASIC_LCASE");
+      assertFalse(collationInfo.isBinaryCollation);
+
+      collationInfo = getInstance().fetchCollationInfo(2);
+      assertEquals(collationInfo.collationName, "UNICODE");
+      assertTrue(collationInfo.isBinaryCollation);
+
+      collationInfo = getInstance().fetchCollationInfo(3);
+      assertEquals(collationInfo.collationName, "UNICODE_CI");
+      assertFalse(collationInfo.isBinaryCollation);
+    }
+
+    @Test
+    public void collationFetchInvalidName() {
+      assertThrows(
```

Comment:
I'd suggest we write test in Scala, so that we can extend `SparkFunSuite` and use convenient test utils like `checkError`

Reply:
I rewrote it to scala. The issue is that `SparkFunSuite` is in `SparkCore` while SparkCollation probably should go in `common`. For now I am using pure `AnyFunSuite` until further advice.

Reply:
Option is to also either move `SparkFunSuite` to common, or move the tests to `core`. I see that `UTF8StringPropertyCheckSuite` suffers from the same issue.